nt/breaksig.c

```diff
@@ -71,7 +71,8 @@ SigHandler (DWORD SigType)
       break;
     case CTRL_CLOSE_EVENT:
       Log (1, "Interrupted by Close");
-      break;
+      binkd_exit = TRUE;
+      return TRUE;
```

Comment:
I do not know much about signal processing on Windows, sorry, so please explain a bit.

What's the difference between `return TRUE` and `return FALSE` here?
Don't we need the same action in other cases (for `CTRL_C_EVENT`, `CTRL_SHUTDOWN_EVENT` etc)?

Reply:
Here's the relevant Microsoft reference:
https://docs.microsoft.com/en-us/windows/console/setconsolectrlhandler

The relevant text is this: "When a console process receives any of the control signals, its handler functions are called on a last-registered, first-called basis until one of the handlers returns TRUE. If none of the handlers returns TRUE, the default handler is called."

The default handler for CTRL_CLOSE_EVENT would immediately terminate the process, which is not what the user (Nick Andre) desired, or how the *nix builds of BinkD behave - instead, we want to set the binkd_exit flag so that we terminate (possibly later) when the process is idle and we can do so gracefully. So the "handler" (SigHandler) needs to return TRUE in this case to inform Windows that we've handled the event/signal and to *not* to continue down the chain of handlers or eventually, the default handler.